vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/FieldBase.java

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.textfield;
+
+import com.vaadin.flow.component.AbstractSinglePropertyField;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.CompositionNotifier;
+import com.vaadin.flow.component.DomEvent;
+import com.vaadin.flow.component.Focusable;
+import com.vaadin.flow.component.HasLabel;
+import com.vaadin.flow.component.HasHelper;
+import com.vaadin.flow.component.HasSize;
+import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.HasValidation;
+import com.vaadin.flow.component.InputNotifier;
+import com.vaadin.flow.component.KeyNotifier;
+import com.vaadin.flow.component.shared.HasClearButton;
+import com.vaadin.flow.component.shared.HasClientValidation;
+import com.vaadin.flow.component.shared.HasThemeVariant;
+import com.vaadin.flow.component.shared.HasTooltip;
+import com.vaadin.flow.data.value.HasValueChangeMode;
+import com.vaadin.flow.dom.Element;
+import com.vaadin.flow.function.SerializableBiFunction;
+import com.vaadin.flow.function.SerializableFunction;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Provides base functionality for combo box related components, such as
+ * {@link TextField}
+ *
+ * @param <TComponent>
+ *            Type of the component that extends from this class
+ * @param <TValue>
+ *            Type of the value of the extending component
+ */
+public abstract class FieldBase<TComponent extends FieldBase<TComponent, TValue>, TValue>
```

Comment:
> Created a new FieldBase class that provides common interfaces / public methods that are shared by all the fields

Reading this, I was kinda expecting this in the shared package and ALL input fields using this as base class (which would be awesome, looking back how V8 had a class that all input fields extended as well)

Reply:
That can be an option as well, although for now I would prefer to keep it in this package.
I'm going to update `TextArea` to also use `FieldBase` as it has most of shared logic.

Reply:
I was more referring to all fields including checkbox, combo box and so on 😅 I can still dream about it tho

Reply:
These could be aligned to some extent, similarly to how we use `@vaadin/field-base` in web components.
However, in some simple cases like `Checkbox` the effort would probably not be worth it.

Reply:
If we want to leave the room open for further refactorings, such as this, we could consider making the class package private for now so splitting/relocating it later wouldn't cause a breaking change

Reply:
> When removing public to make the class package private, I'm getting these compilation errors:

So it seems...not an option then

Reply:
If using package private is not possible. How about a strict renaming of the class to `InternalFieldBase` with a JavaDoc comment? At least then you can always say:  it was internal, don't touch it.

Reply:
@knoobie Good point, thanks. Done.

Reply:
No more getElement dilemma ;)

Reply:
One more thing we could do to enforce this being an internal class is to make the constructors package private. That way no-one should be able to make a component that extends `InternalFieldBase`

Reply:
Missed this comment, sorry about that. Here is a follow up PR: https://github.com/vaadin/flow-components/pull/4410